pyccel/codegen/wrappergen.py

```diff
@@ -0,0 +1,165 @@
+        """
+        current_name_clash_checker = Scope.name_clash_checker
+        ast = self._ast
+        for Wrapper in self._wrapper_types:
+            if self._verbose:
+                print(f">> Building {Wrapper.start_language}-{Wrapper.target_language} interface :: ", self._name)
+
+            Scope.name_clash_checker = name_clash_checkers[Wrapper.start_language.lower()]
+            wrapper = Wrapper(sharedlib_dirpath, verbose = self._verbose)
+
+            ast = wrapper.wrap(ast)
+            self._wrapper_ast.append(ast)
+
+            if errors.has_errors():
+                break
+
+        Scope.name_clash_checker = current_name_clash_checker
+
+    def print(self, dirpath):
+        """
+        Print the wrapper code.
+
+        Print the AST objects generated by a call to the wrap() function.
+
+        Parameters
+        ----------
+        dirpath : str | Path
+            The path to the directory where files should be printed.
+
+        Returns
+        -------
+        list[Path]
+            A list of the source files printed by this function (this is not equivalent
+            to all files printed by this function as headers are excluded).
+        """
+        dirpath = Path(dirpath)
+        files = [dirpath / f'{ast.name}_wrapper.{_extension_registry[Wrapper.start_language.lower()]}'
+                 for ast, Wrapper in zip(self._wrapper_ast, self._wrapper_types)]
+        for i, (filepath, ast, Printer) in enumerate(zip(files, self._wrapper_ast, self._printer_types)):
+            header_ext = _header_extension_registry[Printer.language.lower()]
+
+            if self._verbose:
+                print ('>>> Printing :: ', filepath)
+            printer = Printer(ast.name, verbose=self._verbose)
+
+            # print module
+            code = printer.doprint(ast)
+
+            with open(filepath, 'w', encoding="utf-8") as f:
+                f.write(code)
+
+            # print module header
+            if header_ext is not None:
+                header_filename = dirpath / f'{ast.name}_wrapper.{header_ext}'
+                module_header = ModuleHeader(ast)
+                if self._verbose:
+                    print ('>>> Printing :: ', header_filename)
+                code = printer.doprint(module_header)
+                with open(header_filename, 'w', encoding="utf-8") as f:
+                    f.write(code)
+
+            self._additional_imports[i] = printer.get_additional_imports().copy()
+
+        return files
+
+    def get_additional_imports(self):
+        """
+        Get the objects that were imported by the codeprinters.
+
+        Get the objects that were imported by the codeprinters.
+        These imports may affect the necessary compiler commands.
+
+        Returns
+        -------
+        dict[str, Import]
+            A dictionary mapping the include strings to the import module.
+        """
+        return self._additional_imports
```

Comment:
AFAIU `self._additional_imports` is a list, not a dictionary. Each of its items is a dictionary as described above.

Reply:
You are right. I have double checked and it's use matches up with the fact that it is a list :+1:

Fixed: 6815316b7